--- a/net/minecraft/data/models/model/ModelTemplate.java
+++ b/net/minecraft/data/models/model/ModelTemplate.java
@@ -79,4 +_,8 @@
     public interface JsonFactory {
         JsonObject create(ResourceLocation p_266987_, Map<TextureSlot, ResourceLocation> p_266933_);
     }
+
+    public net.neoforged.neoforge.client.model.generators.ExtendedModelTemplate.Builder extend() {
+        return net.neoforged.neoforge.client.model.generators.ExtendedModelTemplate.Builder.of(this);
+    }
 }
