--- a/net/minecraft/data/models/ItemModelGenerators.java
+++ b/net/minecraft/data/models/ItemModelGenerators.java
@@ -39,9 +_,17 @@
         new ItemModelGenerators.TrimModelData("amethyst", 1.0F, Map.of())
     );
     public final BiConsumer<ResourceLocation, Supplier<JsonElement>> output;
+    public final String modId;
 
+    // Neo: Use mod id sensitive version
+    @Deprecated
     public ItemModelGenerators(BiConsumer<ResourceLocation, Supplier<JsonElement>> p_125082_) {
+        this(p_125082_, ResourceLocation.DEFAULT_NAMESPACE);
+    }
+
+    public ItemModelGenerators(BiConsumer<ResourceLocation, Supplier<JsonElement>> p_125082_, String modId) {
         this.output = p_125082_;
+        this.modId = modId;
     }
 
     public void generateFlatItem(Item p_125089_, ModelTemplate p_125090_) {
